### DISCUSSION TITLE
OpenVSP gemoetry import for Rotors

### DISCUSSION THREAD

**Discussion by tobiokanobi** [byuflowlab/FLOWUnsteady#151] [Q&A] [answered]
I've seen that there is support for OpenVSP and the import of Geometry. It looks like that it is for Fuselage&Wings, but can it be used for rotors, propellers as well (the rotating ones) ?

**Comment by EdoAlvarezR** [comment]
Hi Tobias!

Not currently. However, it is relatively easy to slice the blades generated in OpenVSP and abstract the data that FLOWUnsteady needs. Perhaps we can automate that process in the future (or feel free to add that feature to FLOWUnsteady and send us a pull request =] ). 

Somewhat related to #123

**Comment by cibinjoseph** [comment, marked as answer]
@tobiokanobi There's a .bem (Blade element) file that you can export from OpenVSP which will have the same details required for FLOWUnsteady. You could use it with some processing. The processing is required because FLOWUnsteady uses separate files for each of the chord, pitch, sweep etc. distributions along the blade.

Now that I look at it, in future we could make it compatible with that .bem file too.

**Reply by EdoAlvarezR** [reply]
Good observation.

Even though the tutorials only show examples reading rotor geometries from files, [the core of `generate_rotor(...)`](https://github.com/byuflowlab/FLOWUnsteady/blob/master/src/FLOWUnsteady_rotor.jl#L125-L130) actually takes arrays with the blade distributions and airfoil sections directly. It would be quite simple to write a function `generate_rotor_from_openvsp(...)` that reads the `.bem` file and passes the information to `generate_rotor(...)`.

**Reply by cibinjoseph** [reply]
We could probably add this feature to VSPGeom and have a `generate_rotor` function that utilizes that inside FLOWUnsteady like we did for the other OpenVSP geometry imports.

**Reply by tobiokanobi** [reply]
@cibinjoseph , @EdoAlvarezR thank you all for your fast reply. I'll give the "bem" a try; but I am not sure if I have the time and knowledge to get an automatic processing done :-)
